internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -1895,6 +1895,12 @@ public CompletableFuture<PubSubProduceResult> sendHeartbeat(
       boolean addLeaderCompleteState,
       LeaderCompleteState leaderCompleteState,
       long originTimeStampMs) {
+    if (isClosed) {
+      CompletableFuture<PubSubProduceResult> future = new CompletableFuture<>();
+      future.completedFuture(null);
+      logger.warn("VeniceWriter already closed for partition " + topicPartition.getTopicName());
```

Comment:
Can you tweak the name to be real topic partition id, instead of only topic name?